freeipa/src/main/java/com/sequenceiq/freeipa/converter/instance/InstanceGroupRequestToInstanceGroupConverter.java

```diff
@@ -45,7 +45,7 @@ public InstanceGroup convert(InstanceGroupRequest source, String accountId, Stri
     }
 
     private void addInstanceMetadatas(InstanceGroupRequest request, InstanceGroup instanceGroup, String hostname, String domain) {
-        Set<InstanceMetaData> instanceMetaDataSet = new HashSet<>();
+        Set<InstanceMetaData> instanceMetaDataSet = new LinkedHashSet<>();
```

Comment:
Can you add a unit test, please, that verifies that instances have the right order?

Reply:
Sorry about that. Added a unit test.